kornia/color/lab.py

```diff
@@ -0,0 +1,205 @@
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    # Convert from Linear RGB to sRGB
+    r: torch.Tensor = image[..., 0, :, :]
+    g: torch.Tensor = image[..., 1, :, :]
+    b: torch.Tensor = image[..., 2, :, :]
+
+    rs: torch.Tensor = torch.where(r > 0.04045, torch.pow(((r + 0.055) / 1.055), 2.4), r / 12.92)
+    gs: torch.Tensor = torch.where(g > 0.04045, torch.pow(((g + 0.055) / 1.055), 2.4), g / 12.92)
+    bs: torch.Tensor = torch.where(b > 0.04045, torch.pow(((b + 0.055) / 1.055), 2.4), b / 12.92)
+
+    image_s = torch.stack([rs, gs, bs], dim=-3)
+
+    xyz_im: torch.Tensor = rgb_to_xyz(image_s)
+
+    # normalize for D65 white point
+    xyz_ref_white: torch.Tensor = torch.tensor([0.95047, 1., 1.08883])[..., :, None, None]
+    xyz_ref_white = xyz_ref_white.to(xyz_im.device)
+    xyz_normalized = torch.div(xyz_im, xyz_ref_white)
+
+    xyz_int = torch.where(
+        xyz_normalized > 0.008856,
+        torch.pow(
+            xyz_normalized,
+            1 /
+            3),
+        7.787 *
+        xyz_normalized +
+        4. /
+        29.)
+
+    x: torch.Tensor = xyz_int[..., 0, :, :]
+    y: torch.Tensor = xyz_int[..., 1, :, :]
+    z: torch.Tensor = xyz_int[..., 2, :, :]
+
+    L = (116. * y) - 16.
+    a = 500. * (x - y)
+    b = 200. * (y - z)
+
+    out = torch.stack([L, a, b], dim=-3)
+
+    return out
+
+
+def lab_to_rgb(image: torch.Tensor, eps: float = 1e-12) -> torch.Tensor:
+    r"""Converts a Lab image to RGB.
+
+    Args:
+        image (torch.Tensor): Lab image to be converted to RGB with shape :math:`(*, 3, H, W)`.
+        eps (float): for numerically stability when dividing. Default: 1e-12.
+
+    Returns:
+        torch.Tensor: Lab version of the image with shape :math:`(*, 3, H, W)`.
+
+    Example:
+        >>> input = torch.rand(2, 3, 4, 5)
+        >>> output = lab_to_rgb(input)  # 2x3x4x5
+    """
+    if not isinstance(image, torch.Tensor):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    L: torch.Tensor = image[..., 0, :, :]
+    a: torch.Tensor = image[..., 1, :, :]
+    b: torch.Tensor = image[..., 2, :, :]
+
+    fy = (L + 16.) / 116.
+    fx = (a / 500.) + fy
+    fz = fy - (b / 200.)
+
+    fz = torch.where(fz < 0, torch.zeros_like(fz), fz)
+
+    fxyz = torch.stack([fx, fy, fz], dim=-3)
+
+    # Convert from Lab to XYZ
+    xyz: torch.Tensor = torch.where(fxyz > .2068966, torch.pow(fxyz, 3.0), (fxyz - 4. / 29.) / 7.787)
+
+    # For D65 white point
+    xyz_ref_white: torch.Tensor = torch.tensor([0.95047, 1., 1.08883])[..., :, None, None]
```

Comment:
Should we also support different [illuminants](https://github.com/scikit-image/scikit-image/blob/2db874cc6485186b08d296887a8a91bb2f8865ec/skimage/color/colorconv.py#L419)? or put these values somewhere more central

Reply:
maybe as a dict on the top of this file for now. How would you define the api for different illuminants - `mode='D65'` and so on ? I suggest adding it in a new PR with proper tests.

Reply:
Agree to leave it for another PR if there is demand for it.